File: app/src/main/java/com/example/healthtracker/model/DateConverter.java
package com.example.healthtracker.model;

import androidx.room.TypeConverter;
import java.util.Date;

public class DateConverter {

    @TypeConverter
    public static Date toDate(Long value) {
        return value != null ? new Date(value) : null;
    }

    @TypeConverter
    public static Long toLong(Date date) {
        return date != null ? date.getTime() : null;
    }
}

File: app/src/main/java/com/example/healthtracker/model/DateConverter.kt
package com.example.healthtracker.model

import androidx.room.TypeConverter
import java.util.*

object DateConverter {

    @TypeConverter
    fun toDate(long: Long?): Date? {
        return if(long != null) Date(long) else null
    }

    @TypeConverter
    fun toLong(date: Date?): Long? {
        return date?.time
    }
}